bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java

```diff
@@ -354,7 +354,8 @@ private MessageDTO generateDebugHitMessage(WorkerExecutionContext ctx, LineNumbe
         PackageVarInfo[] packageVarInfoEntries = ctx.programFile.getPackageInfo(ctx.programFile.getEntryPkgName()).
                 getPackageInfoEntries();
         for (PackageVarInfo packVarInfo : packageVarInfoEntries) {
-            if (!packVarInfo.getName().startsWith(META_DATA_VAR_PATTERN)) {
+            // TODO: Need to change the 'contains' logic, if we allow user-defined variable names to have '$'
+            if (!packVarInfo.getName().contains(META_DATA_VAR_PATTERN)) {
```

Comment:
Have you checked with `identifier literals`? Those can contain `$`, right?

Reply:
Yes; Currently, Ballerina does not allow `identifier literals` to have `$`. Hence, I have used this logic for now.

Reply:
Yeah; Currently, it seems we don't have a way to track whether a variable is identifier literal or not at runtime. We can do this by preserving the flags at the runtime. As we need to discuss and decide on this further, for the moment I created an issue (https://github.com/ballerina-platform/ballerina-lang/issues/10336) to track this.

Reply:
Yeah; Currently, it seems we don't have a way to track whether a variable is identifier literal or not at runtime. We may able to do this by preserving the flags at the runtime. As we need to discuss on this further, and identifier literals are seldomly used, for the moment I am skipping this. I created an issue (https://github.com/ballerina-platform/ballerina-lang/issues/10336) to track this.

Reply:
as you mentioned, yes we need a better way to track generated variables, as there is a issue created, will merge this PR